test/features/step_definitions/DungeonMapSteps.cpp

```diff
@@ -163,4 +205,26 @@ THEN( "^The JRect ([0-9.-]+),([0-9.-]+),([0-9.-]+),([0-9.-]+) is now lit$" )
         }
         // printf("\n");
     }
+}
+
+THEN("^I'll have a CDungeonCreationStep$")
+{
+    ScenarioScope<TestCtx> context;
+    // printf("Step is %d\n", context->dungeonCreationStep->m_dwIndex);
+    EXPECT_EQ( context->dungeonCreationStep->m_dwIndex, DUNG_CREATE_STEP_MAKE_ROOM );
+}
+
+THEN("^I'll have a valid area$")
+{
+    ScenarioScope<TestCtx> context;
+    EXPECT_EQ( context->areaChecked, 1 );
+}
+
+THEN("^I'll have a valid N room rect$")
+{
+    ScenarioScope<TestCtx> context;
+    JRect expected = context->room;
+    printf("JRect is <%d, %d, %d, %d>\n", RECT_EXPAND(expected) );
+    EXPECT_EQ( expected.Width(), 6 );
+    EXPECT_EQ( expected.Height(), 2 );
```

Comment:
You are demonstrating that the requested 7x7 room being built from 3,3 in the north direction… gets truncated to 2x6?

Reply:
these values should match vSize passed to GetRoomRect (10,10)